### PR TITLE
Update ClimaCore

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Update ClimaCore and other dependencies

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

File: Project.toml [hunk 1]
```diff
@@ -32,7 +32,6 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
 
 [compat]
 AtmosphericProfilesLibrary = "0.1"
-ClimaCore = "0.10"
```
Comment (simonbyrne) [inline Project.toml:35]: you need to keep this here

**Comment by sriharshakandala** [comment]
bors try

**Comment by charleskawczynski** [comment]
Hi Sriharsha, can you please use the `.dev/up_deps.jl` script? This PR is missing the `ode_compat_examples` environment (and maybe others)

**Comment by sriharshakandala** [comment]
Done.

**Review by valeriabarra — APPROVED** [review]

**Review by charleskawczynski — APPROVED** [review]
The manifest changes look fine to me, I think it's just a matter of getting CI to work and fixing any other issues that may arise

**Comment by charleskawczynski** [comment]
It looks like the latest release increases allocations

**Comment by charleskawczynski** [comment]
Maybe we can add more broad and precise allocation tracking to ClimaCore to keep allocations from from slipping through

**Comment by sriharshakandala** [comment]
I temporarily increased the limits for now.

**Comment by charleskawczynski** [comment]
bors ping

**Pull request merged**